packages/report-server/src/auth/middleware.ts

```diff
@@ -1,20 +1,30 @@
 import { UnauthenticatedError } from '@tupaia/utils';
 import { AccessPolicy } from '@tupaia/access-policy';
+import { getUserIDFromToken } from '@tupaia/auth';
 import { ReportsRequest } from '../types';
 
+const authenticateUser = async (req: ReportsRequest) => {
+  const authHeader = req.headers.authorization || req.headers.Authorization;
+  if (!authHeader) {
+    throw new UnauthenticatedError('No authorization header provided - must be Bearer Auth Header');
+  }
+
+  // Use the user account provided in the auth header if present
+  const tokenUserID = authHeader.startsWith('Bearer') && getUserIDFromToken(authHeader);
+  if (tokenUserID) {
+    return tokenUserID;
+  }
+
+  throw new UnauthenticatedError('Could not authenticate with the provided token');
+}
+
 export const authenticationMiddleware = async (req: ReportsRequest, res, next) => {
   try {
-    const { body, authenticator } = req;
-    const { emailAddress, password } = body;
-    const { user, accessPolicy } = await authenticator.authenticatePassword({
-      emailAddress,
-      password,
-      deviceName: 'reports',
-    });
-    if (user) {
+    const userId = await authenticateUser(req);
+    if (userId) {
+      const { authenticator } = req;
+      const accessPolicy = await authenticator.getAccessPolicyForUser(userId);
```

Comment:
Do we still want to support authenticate by username and password alongside with access token? I thought that since we're only connecting report-server up to psss-server for now it's fine to leave it accepting only access token (similar to meditrak-server). Otherwise we will have to add an extra query parameter `authType` and provide it for every request

Reply:
You make a good argument for removing the user/pass auth

I do however find basic user/pass auth quite useful when testing the report-server. Otherwise we'd need to send an initial request to the /auth endpoints of the meditrak/psss server to get a valid auth token right?

This also shouldn't be too hard to set up using VSCode Rest Client, but if you're testing locally does still require you to have one of those servers up and running...

What do you rekon? I think I lean slightly towards supporting both...